### PR TITLE
Update API to 0.9.0

### LINKED ISSUE #41: Update API for STAC version 0.9.0
### Improvement

There is a new release for 0.9.0 that is coming soon - we should start updating the API to reflect that

### Notes + Context

Here is the [changelog](https://github.com/radiantearth/stac-spec/blob/ad5904d8ccb9cc535ca270650fa90eb66d05ad76/CHANGELOG.md#v090-rc1---2019-12-23) for the upcoming release. There are a bunch of updates to extensions that I don't think are relevant _yet_ for franklin, I've pared down the relevant changes/updates we need to make here.

Here are the relevant changes I think we need to make:
  - The endpoint `/stac` has been merged with `/`
  - The endpoint `/stac/search` is now called `/search`
  - `search` extension renamed to `context` extension. JSON object renamed from `search:metadata` to `context`
  - Removed "next" from the search metadata and query parameter, added POST body and headers to the links for paging support

### PR DESCRIPTION
**Pull request by notthatbreezy**
## Overview

This upgrades the API to STAC version 0.9.0 and upgrades dependencies.

### Checklist

~- [ ] New tests have been added or existing tests have been modified~

### Notes

I made a few decisions here that are worth highlighting.

First, in the issue for the upgrade there is mention about removing the `next` parameter from the `Context` object (formerly `search:metadata`). However, I chose not to do that since we have not yet implemented token based pagination. I updated that issue to note that in the process of implementing pagination that should be removed. It's not a big change to make, but if we do it without implementing pagination as specified in the API spec then there will be no way to support it. So for now, I left it.

Second, I created an example catalog of Sentinel 2 data because I needed a catalog to test with.

### Testing Instructions

- Look over #41 and make sure I made the changes I needed to make
- Clear database
```
docker-compose up -d database
./scripts/dbshell
truncate table collection_items; truncate table collections;
```
- Load Sentinel 2 Sample Catalog:
```
bloop run application -- import --catalog-root s3://rasterfoundry-development-data-us-east-1/catalog.json  --db-user franklin --db-password franklin --db-host localhost --db-name franklin
```
- Start Server: `./scripts/server`
- Test some endpoints:
 - http://localhost:9090/search <- notice the root is gone, this is search results -- datetime filtering, etc. should work here
 - http://localhost:9090/ <- notice I fixed the spec link in here

Closes #41

### CODE REVIEW

File: application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala [hunk 1]
```diff
@@ -9,13 +9,7 @@ import sttp.tapir.json.circe._
...
 object SearchEndpoints {
 
-  val base = endpoint.in("stac")
-
-  val rootCatalog: Endpoint[Unit, Unit, Json, Nothing] =
-    base.get
-      .out(jsonBody[Json])
-      .description("Root Catalog and entrypoint into STAC collections")
-      .name("root")
+  val base = endpoint.in("search")
```
Comment (notthatbreezy) [inline application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala]: `/stac/search` has been removed and only `/search` exists now

File: .scalafmt.conf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 version = 2.4.1
 align = more    // For pretty alignment.
 maxColumn = 100 // For my wide 30" display.
-rewrite.rules = [SortImports]
```
Comment (jisantuc) [inline .scalafmt.conf]: Quick question about this -- the scalafix rule does block sorting, and scalafmt did sorting within lines i thought like 

```diff
- import doobie.{Transactor, ConnectionIO}
+ import doobie.{ConnectionIO, Transactor}
```

is that not right?

Comment (notthatbreezy) [reply .scalafmt.conf]: Oh, I didn't realize that was doing anything -- I'll add it back.

Comment (notthatbreezy) [reply .scalafmt.conf]: I added it back but it didn't change anything - `SortImports` sorts by:

> symbols, lower-case, upper-case.

So in this case they're equivalent

**Pull request merged**